### PR TITLE
Generics cleanup

### PR DESCRIPTION
**Pull request by smcjones**
- Generics propagation was out-of-control. This simplifies generics by housing most of them in a `ClientType` interface where relationships can be validated.
- Got rid of "lazyLoadApp". Instead we declare functions in main.ts and refer to the common frontend methods.
- Added some interfaces to make type referencing easier
- Moved most types to types.ts files.

### CODE REVIEW

File: dv360/src/client.ts [hunk 1]
```diff
@@ -125,23 +108,17 @@ export class Client implements ClientInterface {
...
-    rule: RuleExecutorClass<
-      ClientInterface,
-      RuleGranularity,
-      ClientArgs,
-      Params
-    >,
+    rule: RuleExecutorClass<DisplayVideoClientTypes>,
     settingsArray: ReadonlyArray<string[]>,
   ): ClientInterface {
+    const ruule = this.ruleStore[rule.definition.name] as RuleExecutor<
```
Comment (anaesqueda) [inline dv360/src/client.ts:118]: There is a typo here that will break the code: ruule should be rule

**Review by anaesqueda — CHANGES_REQUESTED** [review]
I found a typo that might break the code, could you please take a look at it? Thanks!

File: dv360/src/client.ts (hunk 1 above)
Comment (smcjones) [reply dv360/src/client.ts]: Thanks for catching - it was an unused variable. It wouldn't have broken the code but definitely shouldn't be there.

File: common/sheet_helpers.ts [hunk 2]
```diff
@@ -744,6 +728,7 @@ export abstract class AppsScriptFrontend<
   displayGlossary() {
     const template = HtmlService.createTemplateFromFile('html/glossary');
     template['rules'] = this.getFrontendDefinitions();
+    template['rules'] = this.getFrontendDefinitions();
```
Comment (smcjones) [inline common/sheet_helpers.ts:731]:
```suggestion
```

**Review by tabrethauer — APPROVED** [review]

**Pull request merged**